### PR TITLE
Maximum value for number field not accepting zero

### LINKED ISSUE #1228: min and max value run time validation not as expected
### Expected behaviour
Must fail as the min value is greater than max value

### Actual behaviour
Accepts even if the min value is greater than max value

### Steps to reproduce the behaviour

Launch test_numbers
enter a min value as 10 and max value as '0'

### PR DESCRIPTION
**Pull request by dwyeradam**
Defaulting to MAX_NUMBER as OR recognises int 0 as false.
Fixes #1228

### CODE REVIEW

File: tests/app/validation/test_number_range_validator.py [hunk 1]
```diff
@@ -195,6 +195,51 @@ def test_manual_max(self):
...
+
+    def test_zero_max(self):
+        answer = { 'max_value': {
+                        "value": 0
+                    },
+                   'label': 'Max Test',
+                   'guidance': '',
+                   'options': [],
+                   'mandatory': False,
+                   'validation':
```
Comment (jonnyshaw89) [inline tests/app/validation/test_number_range_validator.py]: You don't really need to define this here, it just bloats the tests when you could just include `error_messages.py`

File: tests/new_functional/spec/numbers.spec.js [hunk 2]
```diff
@@ -57,4 +57,15 @@ describe('NumericRange', function() {
...
   });
 
+  it('Given a range is set when the maximum is less than the minimum then a run time error should occur', function() {
+    return helpers.openQuestionnaire(number_schema)
+      .then(() => {
+        return browser
+          .setValue(SetMinMax.setMinimum(), '10')
+          .setValue(SetMinMax.setMaximum(), '0')
+          .click(SetMinMax.submit())
+          .getText('#main > h1').should.eventually.contain('Error 500');
```
Comment (jonnyshaw89) [inline tests/new_functional/spec/numbers.spec.js]: Do we want to be testing this sort of things as a functional test?
Seems to me this is more the sort of thing that can be tested with a unit or integration test.

**Review by jonnyshaw89 — DISMISSED** [review]

File: tests/new_functional/spec/numbers.spec.js (hunk 2 above)
Comment (yoganandkunche) [reply tests/new_functional/spec/numbers.spec.js]: @jonnyshaw89 Removed the functional test as we already have a unit test test_number_range_validator

File: tests/app/validation/test_number_range_validator.py (hunk 1 above)
Comment (yoganandkunche) [reply tests/app/validation/test_number_range_validator.py]: Done

**Review by shahi645 — APPROVED** [review]

File: tests/app/validation/test_number_range_validator.py [hunk 3]
```diff
@@ -197,6 +197,44 @@ def test_manual_max(self):
         except ValidationError:
             self.fail("Valid integer raised ValidationError")
 
+
+    def test_zero_max(self):
```
Comment (ajmaddaford) [inline tests/app/validation/test_number_range_validator.py:201]: I think we should also have a test for a min_value of `0` defined in the schema

**Review by ajmaddaford — DISMISSED** [review]

**Review by jcox-dev — APPROVED** [review]
Tests for both changes are in. Looks good to me.

**Pull request merged**